src/storage/mem_mgr.cc

```diff
@@ -0,0 +1,158 @@
+#include <cuda.h>
+#include <cuda_runtime_api.h>
+#include <math.h>
+#include <mxnet/mem_mgr.h>
+#include "../common/cuda_utils.h"
+
+namespace mxnet {
+
+#define CUDA_CALL(func) 					 \
+  {								 \
+    cudaError_t e = (func);     				 \
+    CHECK(e == cudaSuccess || e == cudaErrorCudartUnloading)     \
+        << "CUDA: " << cudaGetErrorString(e);                    \
+  }
```

Comment:
Please use "static inline" instead of macro (#define). You can google "c++ static inline macro" to learn more.